includes/Mapper/class-contactmapper.php

```diff
@@ -0,0 +1,198 @@
+		$options = get_option( 'omnisend_lifterlms_options' );
+		$contact = new Contact();
+
+		$contact->set_email( $mapped_fields['email_address'] );
+		$contact->set_phone( $mapped_fields['llms_phone'] );
+		$contact->set_first_name( $mapped_fields['first_name'] ?? '' );
+		$contact->set_last_name( $mapped_fields['last_name'] ?? '' );
+		$contact->set_postal_code( $mapped_fields['llms_billing_zip'] ?? '' );
+		$contact->set_address( $mapped_fields['llms_billing_address_1'] . ' ' . $mapped_fields['llms_billing_address_2'] ?? '' );
+		$contact->set_state( $mapped_fields['llms_billing_state'] ?? '' );
+		$contact->set_country( $mapped_fields['llms_billing_country'] ?? '' );
+		$contact->set_city( $mapped_fields['llms_billing_city'] ?? '' );
+
+		$contact->set_welcome_email( true );
+
+		if ( ! isset( $options['filter_lms_consent_setting'] ) ) {
+			if ( isset( $mapped_fields['llmsconsentEmail'] ) ) {
+				$contact->set_email_subscriber();
+				$contact->set_email_consent( self::CONSENT_PREFIX );
+			} else {
+				$contact->set_email_consent( self::CONSENT_PREFIX );
+				$contact->set_email_unsubscriber();
+			}
+
+			if ( isset( $mapped_fields['llmsconsentPhone'] ) ) {
+				$contact->set_phone_consent( self::CONSENT_PREFIX );
+				$contact->set_phone_subscriber();
+			} else {
+				$contact->set_phone_consent( self::CONSENT_PREFIX );
+				$contact->set_phone_unsubscriber();
+			}
+		} else {
+			$contact->set_email_consent( self::CONSENT_PREFIX );
+			$contact->set_email_subscriber();
+
+			$contact->set_phone_consent( self::CONSENT_PREFIX );
+			$contact->set_phone_subscriber();
+		}
+
+		$contact->add_tag( self::CUSTOM_PREFIX );
+
+		return $contact;
+	}
+
+	/**LifterLMSAddon
+	 * Update Contact object
+	 *
+	 * @param array  $mapped_fields
+	 *
+	 * @return Contact object
+	 */
+	public function update_contact( array $mapped_fields ): Contact {
+		$options = get_option( 'omnisend_lifterlms_options' );
+		$contact = new Contact();
+
+		$current_user = wp_get_current_user();
+		$user_email   = $current_user->user_email;
+
+		$contact->set_email( $user_email );
+		$contact->set_phone( $mapped_fields['llms_phone'] );
+		$contact->set_first_name( $mapped_fields['first_name'] ?? '' );
+		$contact->set_last_name( $mapped_fields['last_name'] ?? '' );
+		$contact->set_postal_code( $mapped_fields['llms_billing_zip'] ?? '' );
+		$contact->set_address( $mapped_fields['llms_billing_address_1'] . ' ' . $mapped_fields['llms_billing_address_2'] ?? '' );
+		$contact->set_state( $mapped_fields['llms_billing_state'] ?? '' );
+		$contact->set_country( $mapped_fields['llms_billing_country'] ?? '' );
+		$contact->set_city( $mapped_fields['llms_billing_city'] ?? '' );
+		$contact->set_welcome_email( true );
+
+		if ( ! isset( $options['filter_lms_consent_setting'] ) ) {
+			if ( isset( $mapped_fields['llmsconsentEmail'] ) ) {
+				$contact->set_email_subscriber();
+				$contact->set_email_consent( self::CONSENT_PREFIX );
```

Comment:
I wrote this in other plugin same opt_in was mentioned:
It's working fine since we updated omnisend SDK plugin by adding set_email_subscriber, set_email_unsubscriber and set_phone_subscriber, set_phone_unsubscriber.
Setting email_opt_in or phone_opt_in now sets email_status and phone_status. to 'subscribed' so in this case using only set_email_subscriber and set_email_unsubscriber sets these statuses and user never will be nonsubscriber unless we will not set these statuses (but in these forms we want to set them since client can update them everytime form is submited)